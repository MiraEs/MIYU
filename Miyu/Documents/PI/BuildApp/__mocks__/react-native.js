/* eslint-disable react/no-multi-comp */
import {
	createElement,
	Component,
} from 'react';
import PropTypes from 'prop-types';
const ReactNative = jest.genMockFromModule('react-native');

ReactNative.StyleSheet = {
	create: function create(styles) {
		return styles;
	},
	flatten: function flatten(...styles) {
		return Object.assign({}, ...styles);
	},
};

ReactNative.PixelRatio = {
	get() {
		return 1;
	},
};

ReactNative.Alert = {
	alert: jest.fn(),
};

ReactNative.StatusBar = {
	setBarStyle: jest.fn(),
	setHidden: jest.fn(),
};

ReactNative.Linking = {
	openURL: jest.fn(),
};

ReactNative.NativeModules = {
	PhotoUploader: {},
	SafariViewManager: {},
	UIManager: {
		measure: jest.fn(),
	},
	MarketingCloud: {
		trackAction: jest.fn(),
		trackState: jest.fn(),
	},
	LocalyticsManager: {
		tagScreen: jest.fn(),
		tagEvent: jest.fn(),
		tagCustomerLoggedOut: jest.fn(),
	},
	Instabug: {
		log: jest.fn(),
	},
	BNDevice: {},
	RNApplePayManager: {
		canMakePayments: jest.fn(() => Promise.resolve(true)),
		canMakePaymentsUsingNetworks: jest.fn(() => Promise.resolve(true)),
		paymentRequest: jest.fn(),
		paymentRequestDidFinish: jest.fn(),
		updateShippingMethods: jest.fn(),
		selectShippingMethod: jest.fn(),
		authorizedPayment: jest.fn(),
	},
};

ReactNative.requireNativeComponent = jest.fn();

ReactNative.NetInfo = {
	isConnected: {
		fetch: jest.fn(() => ({ done: jest.fn() })),
		addEventListener: jest.fn(),
		removeEventListener: jest.fn(),
	},
};

ReactNative.AppState = {
	addEventListener: jest.fn(),
};

ReactNative.CameraRoll = {
	getPhotos: jest.fn(() => Promise.resolve([])),
};

ReactNative.PushNotificationIOS = {
	setApplicationIconBadgeNumber: jest.fn(),
};

ReactNative.Keyboard = {
	dismiss: jest.fn(),
};

ReactNative.findNodeHandle = jest.fn();

ReactNative.ViewPropTypes = {
	style: jest.fn(),
};

ReactNative.NativeEventEmitter = class NativeEventEmitter {
	addListener = jest.fn(() => ({ remove: jest.fn() }));
};

class FlatList extends Component {
	render() {
		return createElement('FlatList', this.props);
	}
}

class SectionList extends Component {
	render() {
		return createElement('SectionList', this.props);
	}
}

class View extends Component {
	static propTypes = {
		children: PropTypes.any,
		style: () => {},
	};
	render() {
		return createElement('View', this.props, this.props.children);
	}
}

class ScrollView extends Component {
	static propTypes = {
		children: PropTypes.any,
		style: () => {},
	};
	render() {
		return createElement('ScrollView', this.props, this.props.children);
	}
}

class Switch extends Component {
	render() {
		return createElement('Switch');
	}
}

class Text extends Component {
	static propTypes = {
		children: PropTypes.any,
		style: () => {},
	};
	render() {
		return createElement('Text', this.props, this.props.children);
	}
}

class TextInput extends Component {
	static propTypes = {
		children: PropTypes.any,
	};
	render() {
		return createElement('TextInput', this.props, this.props.children);
	}
}

class TouchableOpacity extends Component {
	static propTypes = {
		children: PropTypes.any,
	};
	render() {
		return createElement('TouchableOpacity', this.props, this.props.children);
	}
}

class TouchableWithoutFeedback extends Component {
	static propTypes = {
		children: PropTypes.any,
	};
	render() {
		return createElement('TouchableWithoutFeedback', this.props, this.props.children);
	}
}

class ToolbarAndroid extends Component {
	static propTypes = {
		children: PropTypes.any,
	};
	render() {
		return createElement('ToolbarAndroid', this.props, this.props.children);
	}
}

class Image extends Component {
	static propTypes = {
		children: PropTypes.any,
		style: () => {},
	};
	render() {
		return createElement('Image', this.props, this.props.children);
	}
}

class Icon extends Component {
	static propTypes = {
		children: PropTypes.any,
	};
	render() {
		return createElement('Icon', this.props, this.props.children);
	}
}

class ActivityIndicator extends Component {
	static propTypes = {
		children: PropTypes.any,
	};
	render() {
		return createElement('ActivityIndicator', this.props, this.props.children);
	}
}

ReactNative.Platform = {
	OS: 'ios',
	select: jest.fn((options) => options.ios),
};

ReactNative.InteractionManager = {
	runAfterInteractions: jest.fn(),
};

ReactNative.ActionSheetIOS = {
	showActionSheetWithOptions: jest.fn(),
};

class ListView extends Component {
	DataSource() {
		return {
			cloneWithRows: jest.fn(),
			cloneWithRowsAndSections: jest.fn(),
		};
	}
}

class AppRegistry {
	registerComponent() {}
}

class TouchableHighlight extends Component {
	static propTypes = {
		children: PropTypes.any,
	};
	render() {
		return createElement('TouchableHighlight', this.props, this.props.children);
	}
}
class BackHandler extends Component {
	static addEventListener() {}
	static exitApp() {}
	static removeEventListener() {}
}

class TouchableNativeFeedback extends Component {
	static Ripple() {
	}
}

class Navigator extends Component {
	render() {
	}
}

class RefreshControl extends Component {
	render() {
		return createElement('RefreshControl', this.props);
	}
}

class LayoutAnimation extends Component {
	static configureNext() {
	};

	static Presets() {
		return {
			easeInEaseOut: jest.fn(),
		};
	};
}

class Dimensions {
	static get() {
		return 0;
	}
}

class Animated extends Component {

	static Value() {
		return {
			getLayout: () => {},
			interpolate: () => {},
			setOffset: () => {},
			setValue: () => {},
		};
	}

	static ValueXY() {
		return {
			getLayout: () => {},
			interpolate: () => {},
			setOffset: () => {},
			setValue: () => {},
		};
	}

	static event() {}
}

class AnimatedView extends Component {
	render() {
		return createElement('AnimatedView');
	}
}

class AnimatedImage extends Component {

}

class AnimatedText extends Component {

}

Animated.View = AnimatedView;

Animated.Image = AnimatedImage;

Animated.Text = AnimatedText;

class PanResponder {
	static create() {
		return {panHandlers: {}};
	}
}

global.__fbBatchedBridgeConfig = {
	remoteModuleConfig: [],
	localModulesConfig: [],
};

ReactNative.View = View;
ReactNative.ScrollView = ScrollView;
ReactNative.FlatList = FlatList;
ReactNative.ListView = ListView;
ReactNative.Text = Text;
ReactNative.TextInput = TextInput;
ReactNative.TouchableOpacity = TouchableOpacity;
ReactNative.TouchableHighlight = TouchableHighlight;
ReactNative.TouchableWithoutFeedback = TouchableWithoutFeedback;
ReactNative.ToolbarAndroid = ToolbarAndroid;
ReactNative.Image = Image;
ReactNative.AppRegistry = AppRegistry;
ReactNative.Icon = Icon;
ReactNative.ActivityIndicator = ActivityIndicator;
ReactNative.BackHandler = BackHandler;
ReactNative.TouchableNativeFeedback = TouchableNativeFeedback;
ReactNative.Navigator = Navigator;
ReactNative.RefreshControl = RefreshControl;
ReactNative.LayoutAnimation = LayoutAnimation;
ReactNative.Dimensions = Dimensions;
ReactNative.Animated = Animated;
ReactNative.PanResponder = PanResponder;
ReactNative.SectionList = SectionList;
ReactNative.Switch = Switch;

module.exports = ReactNative;
