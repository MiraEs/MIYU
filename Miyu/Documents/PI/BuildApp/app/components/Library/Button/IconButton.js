'use strict';
import React, {
	Component,
} from 'react';
import PropTypes from 'prop-types';
import ReactNative, {
	ViewPropTypes,
} from 'react-native';
import {
	Button,
	Text,
} from 'BuildLibrary';
import Icon from 'react-native-vector-icons/Ionicons';
import styles from '../../../lib/styles';


const componentStyles = ReactNative.StyleSheet.create({
	layout: {
		flex: 1,
		flexDirection: 'row',
		alignItems: 'center',
		marginRight: styles.measurements.gridSpace1,
	},
	text: {
		marginHorizontal: styles.measurements.gridSpace1,
	},
});

class IconButton extends Component {

	getIconSize = () => {
		const { iconSize } = this.props;
		if (iconSize === 'small') {
			return 16;
		} else if (iconSize === 'large') {
			return 32;
		}
		return 25;
	};

	renderText() {
		const { size, text, textColor } = this.props;

		if (text) {
			return (
				<Text
					numberOfLines={1}
					style={componentStyles.text}
					color={textColor}
					size={size}
					weight="bold"
					lineHeight={false}
					textAlign="center"
				>
					{text}
				</Text>
			);
		}
	}

	render() {
		const { borders, color, iconName, iconStyle, isLoading, onPress, size, style, textColor, trackAction, trackContextData } = this.props;

		return (
			<Button
				borders={borders}
				color={color}
				textColor={textColor}
				isLoading={isLoading}
				onPress={onPress}
				size={size}
				style={style}
				trackAction={trackAction}
				trackContextData={trackContextData}
				accessibilityLabel={this.props.accessibilityLabel}
			>
				<ReactNative.View style={componentStyles.layout}>
					<Icon
						color={styles.colors[textColor]}
						name={iconName}
						size={this.getIconSize()}
						style={iconStyle}
					/>
					{this.renderText()}
				</ReactNative.View>
			</Button>
		);
	}
}

IconButton.propTypes = {
	onPress: PropTypes.func.isRequired,
	color: PropTypes.oneOf([
		'error',
		'facebookBlue',
		'primary',
		'white',
	]),
	borders: PropTypes.bool,
	iconName: PropTypes.string.isRequired,
	iconSize: PropTypes.oneOf([
		'small',
		'regular',
		'large',
	]),
	iconStyle: ViewPropTypes.style,
	isLoading: PropTypes.bool,
	size: PropTypes.oneOf([
		'small',
		'regular',
	]),
	style: ViewPropTypes.style,
	text: PropTypes.any,
	textColor: PropTypes.oneOf([
		'primary',
		'secondary',
		'white',
		'accent',
	]),
	trackAction: PropTypes.string,
	trackContextData: PropTypes.object,
	accessibilityLabel: PropTypes.string.isRequired,
};

IconButton.defaultProps = {
	color: 'primary',
	borders: true,
	iconSize: 'regular',
	isLoading: false,
	size: 'regular',
	textColor: 'secondary',
};

export default IconButton;
