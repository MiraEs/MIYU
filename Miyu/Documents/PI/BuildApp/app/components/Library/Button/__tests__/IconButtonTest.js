'use strict';
jest.unmock('react-native');
jest.mock('../../../../../app/store/configStore', () => ({}));
jest.mock('BuildNative');
jest.mock('BuildLibrary');

import 'react-native';
import React from 'react';
import IconButton from '../IconButton';

const defaultProps = {
	onPress: jest.fn(),
	trackAction: 'trackAction',
	accessibilityLabel: 'accessibilityLabel',
	iconName:'ios-alert',
	children : [],
};

describe('BuildLibrary IconButton', () => {
	it('should render correctly', () => {
		const tree = require('react-test-renderer').create(
			<IconButton {...defaultProps} />
		).toJSON();
		expect(tree).toMatchSnapshot();
	});

});
