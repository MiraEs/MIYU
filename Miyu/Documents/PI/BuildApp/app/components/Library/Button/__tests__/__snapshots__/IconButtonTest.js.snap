// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BuildLibrary IconButton should render correctly 1`] = `
<Button
  accessibilityLabel="accessibilityLabel"
  borders={true}
  color="primary"
  isLoading={false}
  onPress={[Function]}
  size="regular"
  style={undefined}
  textColor="secondary"
  trackAction="trackAction"
  trackContextData={undefined}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "marginRight": 8,
      }
    }
  >
    <Icon
      color="#63666A"
      name="ios-alert"
      size={25}
      style={undefined}
    />
  </View>
</Button>
`;
